File: algorithms/1_easy/W/01518_water_bottles/solution.py
class Solution:
    def numWaterBottles(self, num_bottles: int, num_exchange: int) -> int:
        count = num_bottles
        while num_bottles >= num_exchange:
            count += num_bottles // num_exchange
            num_bottles = num_bottles // num_exchange + num_bottles % num_exchange

        return count

File: algorithms/1_easy/W/01518_water_bottles/solution.rs
impl Solution {
    pub fn num_water_bottles(mut num_bottles: i32, num_exchange: i32) -> i32 {
        let mut count = num_bottles;
        while num_bottles >= num_exchange {
            count += num_bottles / num_exchange;
            num_bottles = num_bottles / num_exchange + num_bottles % num_exchange;
        }

        count
    }
}
